crates/revm/Cargo.toml

```diff
@@ -22,8 +22,8 @@ revm-precompile = { path = "../precompile", version = "2.2.0", default-features
 auto_impl = { version = "1.1", default-features = false }
 
 # Optional
-serde = { version = "1.0", features = ["derive", "rc"], optional = true }
-serde_json = { version = "1.0", features = ["preserve_order"], optional = true }
+serde = { version = "1.0", default-features = false, features = ["derive", "rc"], optional = true }
+serde_json = { version = "1.0", default-features = false, features = ["alloc", "preserve_order"], optional = true }
```

Comment:
@rakita how critical is having `preserve_order` here?

Reply:
It seems not at all critical, I thought it is there to format output, but after checking, this is not the case. I am okay with removing it.

Reply:
Nice, thanks!

Reply:
Just curious why would you need binary to be `no_std` compatible?

Reply:
My need (and in this case @CeciliaZ030 's as well) is not the binary per se, but rather revm+serde. We want no-std revm to make ZK proofs for it with [powdr](https://github.com/powdr-labs/powdr) , which works fine. However it would be great to also be able to enable the `serde` feature in revm, in order to have better UX in message passing between the host and the guest when making ZK proofs.

Reply:
Makes sense, I don't focus a lot of my attention on `revme` (binary) as it is used only for testing, was curious to see if I need to revise that view.

Okay, I just noticed that `preserve_order` is in both revm and revme. `preserved_order` is nice in `revm` as it formats trace json output that is easier to read, still okay with removing it.

Can you make a PR?

Reply:
Sure! We can also wait to see if https://github.com/serde-rs/json/pull/1101 gets merged, then we don't need to remove it.

Reply:
Your call, if this blocks you or you can use this right now it would be best to make PR